test/bdd/pkg/edv/edv_steps.go

```diff
@@ -49,71 +73,325 @@ func (e *Steps) createDataVault(vaultID, expectedVaultLocation string) error {
+		return "", "", "", "", fieldNotFoundError("ciphertext", jweDocType)
+	}
+
+	tagFieldValue, found := jweFields["tag"]
+	if !found {
+		return "", "", "", "", fieldNotFoundError("tag", jweDocType)
+	}
+
+	return protectedFieldValue, ivFieldValue, ciphertextFieldValue, tagFieldValue, nil
+}
+
+func verifyFieldsAreEqual(retrievedProtectedFieldValue, expectedProtectedFieldValue, retrievedIVFieldValue,
+	expectedIVFieldValue, retrievedCiphertextFieldValue, expectedCiphertextFieldValue, retrievedTagFieldValue,
+	expectedTagFieldValue string) error {
+	if retrievedProtectedFieldValue != expectedProtectedFieldValue {
+		return unexpectedValueError(expectedProtectedFieldValue, retrievedProtectedFieldValue)
+	}
+
+	if retrievedIVFieldValue != expectedIVFieldValue {
+		return unexpectedValueError(expectedIVFieldValue, retrievedIVFieldValue)
+	}
+
+	if retrievedCiphertextFieldValue != expectedCiphertextFieldValue {
+		return unexpectedValueError(expectedCiphertextFieldValue, retrievedCiphertextFieldValue)
+	}
+
+	if retrievedTagFieldValue != expectedTagFieldValue {
+		return unexpectedValueError(expectedTagFieldValue, retrievedTagFieldValue)
+	}
+
+	return nil
+}
+
+func verifyStructuredDocsAreEqual(decryptedDoc, expectedDoc operation.StructuredDocument) error {
+	if decryptedDoc.ID != expectedDoc.ID {
+		return unexpectedValueError(expectedDoc.ID, decryptedDoc.ID)
+	}
+
+	expectedCreatedValue, decryptedCreatedValue, err := getMetaFieldValues(expectedDoc, decryptedDoc)
+	if err != nil {
+		return err
+	}
+
+	expectedMessageValue, decryptedMessageValue, err := getContentFieldValues(expectedDoc, decryptedDoc)
+	if err != nil {
+		return err
+	}
+
+	if decryptedCreatedValue != expectedCreatedValue {
+		return unexpectedValueError(expectedCreatedValue, decryptedCreatedValue)
+	}
+
+	if decryptedMessageValue != expectedMessageValue {
+		return unexpectedValueError(expectedMessageValue, decryptedMessageValue)
 	}
 
 	return nil
 }
+
+func getContentFieldValues(expectedDoc, decryptedDoc operation.StructuredDocument) (string, string, error) {
+	expectedMessageFieldInContent, found := expectedDoc.Content["message"]
+	if !found {
+		return "", "", fieldNotFoundError("message", "expected Structured Document")
+	}
+
+	expectedMessageFieldInContentString, ok := expectedMessageFieldInContent.(string)
+	if !ok {
+		return "", "", unableToAssertAsStringError("message")
+	}
+
+	decryptedMessageFieldInContent, found := decryptedDoc.Content["message"]
+	if !found {
+		return "", "", fieldNotFoundError("message", "decrypted Structured Document")
+	}
+
+	decryptedMessageFieldInContentString, ok := decryptedMessageFieldInContent.(string)
+	if !ok {
+		return "", "", unableToAssertAsStringError("message")
+	}
+
+	return expectedMessageFieldInContentString, decryptedMessageFieldInContentString, nil
+}
+
+func getMetaFieldValues(expectedDoc, decryptedDoc operation.StructuredDocument) (string, string, error) {
+	expectedCreatedFieldInMeta, found := expectedDoc.Meta["created"]
```

Comment:
use a const for `created` key in this function

Reply:
Done